provisioning/baremetal_pod.go

```diff
@@ -586,13 +584,21 @@ func newMetal3Deployment(targetNamespace string, images *Images, config *metal3i
 		}
 	}
 	k8sAppLabel := metal3AppName
+	apiLabelValue := ""
 	for k, v := range selector.MatchLabels {
 		if k == "k8s-app" {
 			k8sAppLabel = v
-			break
 		}
+		if k == apiLabelKey {
+			apiLabelValue = v
+		}
+	}
+	podSpecLabels := map[string]string{
+		"k8s-app":    k8sAppLabel,
+		apiLabelKey:  apiLabelValue,
```

Comment:
This means that we will always have an `api` label (with an empty value when we are creating a new Deployment). I think it would be better if we just didn't add that label at all unless it is necessary for the selector.

Reply:
Ok. Cool!